Test/test2/LambdaExt.bpl

```diff
@@ -67,9 +67,6 @@ procedure Quantifiers() {
 }
 
 procedure FreeVariables() {
```

Comment:
Here is another test you could add. It checks that free variables in triggers are substituted appropriately.

```
function g(int,int) : int;

procedure Triggers'(w: int, w': int) {
  var a,b : [int]bool;
  a := (lambda x:int :: (forall u:int :: { g(u,w) } x == g(u,w)));
  b := (lambda y:int :: (forall v:int :: { g(v,w') } y == g(v,w')));
  assert w == w' ==> a == b;
  b := (lambda y:int :: (forall v:int :: y == g(v,w')));
  assert w == w' ==> a == b; // should fail because triggers are different
}
```

Reply:
Thanks!